content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,361 +36,581 @@ are covered here as well as the `System.Diagnostics` API.
+namespace otel
+{
+    public class DiceController : ControllerBase
+    {
+        private ILogger<DiceController> logger;
+
+        public DiceController(ILogger<DiceController> logger)
+        {
+            this.logger = logger;
+        }
+
+        [HttpGet("/rolldice")]
+        public List<int> RollDice(string player, int? rolls)
+        {
+            if(!rolls.HasValue)
+            {
+                logger.LogError("Missing rolls parameter");
+                throw new HttpRequestException("Missing rolls parameter", null, HttpStatusCode.BadRequest);
+            }
+                
+            var result = new Dice(1, 6).rollTheDice(rolls.Value);
+
+            if (string.IsNullOrEmpty(player))
+            {
+                logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
+            }
+            else
+            {
+                logger.LogInformation("{player} is rolling the dice: {result}", player, result);
+            }
+
+            return result;
+        }
+    }
+}
+```
 
-var serviceName = "MyServiceName";
-var serviceVersion = "1.0.0";
+Replace the content of the program.cs file with the following code:
 
-using var tracerProvider = Sdk.CreateTracerProviderBuilder()
-    .AddSource(serviceName)
-    .ConfigureResource(resource =>
-        resource.AddService(
-          serviceName: serviceName,
-          serviceVersion: serviceVersion))
-    .AddConsoleExporter()
-    .Build();
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```sh
+dotnet build
+dotnet run
+```
 
-// ...
+You should get a list of 12 numbers in your browser window, for example:
+
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
 ```
 
-This is also where you can configure instrumentation libraries.
+## Manual instrumentation setup
+
+### Dependencies
 
-Note that this sample uses the Console Exporter. If you are exporting to another
-endpoint, you'll have to use a different exporter.
+Install the following OpenTelemetry NuGet packages:
 
-#### ASP.NET Core
+[OpenTelemetry.Exporter.Console](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
 
-To start tracing in an ASP.NET Core-based app, use the OpenTelemetry extensions
-for ASP.NET Core setup.
+[OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting)
 
-First, ensure that you have the right packages:
 
 ```sh
-dotnet add package OpenTelemetry
-dotnet add package OpenTelemetry.Extensions.Hosting
 dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package OpenTelemetry.Extensions.Hosting
 ```
 
-Then you can install the Instrumentation package
+For ASP.NET Core-based applications, also install the AspNetCore instrumentation package
```

Comment:
this is not used in the rest of the code... missed to add?

Reply:
Yes, I forgot to include it in the "Initialize the SDK" section.

I initially thought it wasnt needed but I noticed the code couldnt generate traces until I included the package.